tests/base.py

```diff
@@ -0,0 +1,108 @@
+import os
+import pathlib
+import random
+import subprocess
+import tempfile
+import time
+import unittest
+
+import rpyc
+
+from .utils import debug_target
+
+COVERAGE_DIR = os.getenv("COVERAGE_DIR", "")
+GEF_PATH = pathlib.Path(os.getenv("GEF_PATH", "gef.py")).absolute()
+RPYC_GEF_PATH = GEF_PATH.parent / "scripts/remote_debug.py"
+RPYC_HOST = "localhost"
+RPYC_PORT = 18812
+RPYC_SPAWN_TIME = 1.0
+RPYC_MAX_REMOTE_CONNECTION_ATTEMPTS = 5
+
+
+class RemoteGefUnitTestGeneric(unittest.TestCase):
+    """
+    The base class for GEF test cases. This will create the `rpyc` environment to programmatically interact with
+    GDB and GEF in the test.
+    """
+
+    def setUp(self) -> None:
+
+        attempt = RPYC_MAX_REMOTE_CONNECTION_ATTEMPTS
+        while True:
+            try:
+                #
+                # Port collisions can happen, allow a few retries
+                #
+                self._coverage_file = None
+                self.__setup()
+                break
+            except ConnectionRefusedError:
+                attempt -= 1
+                if attempt == 0:
+                    raise
+                time.sleep(0.2)
+                continue
+
+        self._gdb = self._conn.root.gdb
+        self._gef = self._conn.root.gef
+        return super().setUp()
+
+    def __setup(self):
+        if not hasattr(self, "_target"):
+            setattr(self, "_target", debug_target("default"))
+        else:
+            assert isinstance(self._target, pathlib.Path)  # type: ignore pylint: disable=E1101
+            assert self._target.exists()  # type: ignore pylint: disable=E1101
+
+        #
+        # Select a random tcp port for rpyc
+        #
+        self._port = random.randint(1025, 65535)
```

Comment:
Why random? Is this to avoid collisions for concurrently running tests? I like to minimize randomness to improve reproducibility.

Reply:
Because of forked tests, incremental ports just don't work as I intended so I went with this for me.
There is a retry in case the port is already bound